### DISCUSSION TITLE
Do TransientProperties persist within Bookmarks

### DISCUSSION THREAD

**Discussion by defunky** [elsa-workflows/elsa-core#5756] [Q&A]
As I understand TransientProperties are properties only exist while the activity executes and are not persisted. If I create bookmark with a callback method, can I access these properties within the callback method or am I stuck using regular Properties?

**Comment by glime-ai** [comment]
Transient properties, as the name suggests, are temporary and only exist during the execution of an activity. They are not persisted with the workflow state. When you create a bookmark with a callback method, the transient properties that were set during the activity execution will not be available in the callback method because the workflow may have been persisted and resumed at a later time, and the transient properties would not have been saved.

If you need to access certain values within the callback method, you should use regular properties that are persisted with the workflow state. These properties will be available when the workflow is resumed from a bookmark.

This comment was generated by [Glime](https://glimelab.ai/?utm_source=dRbvTb7IhCSYiI&utm_medium=github).
